### DISCUSSION TITLE
Save user consent to terms and conditions checkbox at checkout

### DISCUSSION THREAD

**Discussion by YuriBendall** [woocommerce/woocommerce#54090] [Checkout Flow]
In the checkout page using woocommerce there is the option to add a terms and conditions and privacy policy checkbox and make it required to be checked off before purchase. I want to record this consent that the user agreed to the terms and privacy policy in the order details in WordPress orders and also anywhere the user may see the order such as account orders and order emails. I want to show proof that the user agreed to the terms and privacy policies and be in compliance with PIPEDA. If there are also options to add more checkboxes for additional requirements and record that as well that would be great.

I have tried PHP code online and plugins such as ACF (although I think there is a dispute going on with WordPress), and neither worked. Do you have any advice or ways to make this happen?

Also, if it helps I am using the Assembler WordPress theme with Woocommerce.

**Comment by mikejolley** [comment]
As it stands, the terms and conditions checkbox will block checkout, but it will not post or store to the order or user profile. If this is a requirement I'd recommend posting a feature request with full details here https://github.com/woocommerce/woocommerce/issues/new?template=2-enhancement.yml

If you're using the block based checkout, you could either:

1. Store some meta data on the order (using custom code) when an order is placed. We'd be assuming that if the order is being placed, the user must have accepted the terms in order for it to go through
2. Register a custom field using our API. These values are persisted to the order.

For the second option, see:

https://developer.woocommerce.com/docs/cart-and-checkout-additional-checkout-fields/

However checkboxes would require some additional validation because of this limitation:

> The checkbox field type does not have any specific options, however required will always be false for a checkbox field.

Making checkboxes a requirement without extra custom validation is coming in a future update.

Thanks
